src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs

```diff
@@ -492,23 +492,31 @@ private OpenApiMediaType CreateResponseMediaType(ModelMetadata modelMetadata, Sc
             typeof(RequiredAttribute)
         };
 
-        private static readonly Dictionary<string, string> ResponseDescriptionMap = new Dictionary<string, string>
+        private static readonly IReadOnlyCollection<KeyValuePair<string, string>> ResponseDescriptionMap = new[]
         {
-            { "1\\d{2}", "Information" },
-            { "2\\d{2}", "Success" },
-            { "304", "Not Modified" },
-            { "3\\d{2}", "Redirect" },
-            { "400", "Bad Request" },
-            { "401", "Unauthorized" },
-            { "403", "Forbidden" },
-            { "404", "Not Found" },
-            { "405", "Method Not Allowed" },
-            { "406", "Not Acceptable" },
-            { "408", "Request Timeout" },
-            { "409", "Conflict" },
-            { "4\\d{2}", "Client Error" },
-            { "5\\d{2}", "Server Error" },
-            { "default", "Error" }
+           new KeyValuePair<string, string>("1\\d{2}", "Information"),
+
+            new KeyValuePair<string, string>("201", "Created"),
+            new KeyValuePair<string, string>("202", "Accepted"),
+            new KeyValuePair<string, string>("204", "No Content"),
+            new KeyValuePair<string, string>("2\\d{2}", "Success"),
+
+            new KeyValuePair<string, string>("304", "Not Modified"),
+            new KeyValuePair<string, string>("3\\d{2}", "Redirect"),
+
+            new KeyValuePair<string, string>("400", "Bad Request"),
+            new KeyValuePair<string, string>("401", "Unauthorized"),
+            new KeyValuePair<string, string>("403", "Forbidden"),
+            new KeyValuePair<string, string>("404", "Not Found"),
+            new KeyValuePair<string, string>("405", "Method Not Allowed"),
+            new KeyValuePair<string, string>("406", "Not Acceptable"),
+            new KeyValuePair<string, string>("408", "Request Timeout"),
+            new KeyValuePair<string, string>("409", "Conflict"),
+            new KeyValuePair<string, string>("429", "Too Many Requests"),
+            new KeyValuePair<string, string>("4\\d{2}", "Client Error"),
+
+            new KeyValuePair<string, string>("5\\d{2}", "Server Error"),
+            new KeyValuePair<string, string>("default", "Error")
```

Comment:
@MikeFH - I'm happy to include the specific descriptions for more of the status codes as you've done here. But, I'd prefer to stick with the more simplified `Dictionary<,>` initializers. I get the desire to make it readonly it's private to the class and IMO highly unlikely that anyting will try to mutate it. So, I'd opt for the reduced boilerplate/increased readability instead.

Reply:
If you're happy to revert this and then resolve the conflicts, I'll go ahead and merge 👍

Reply:
Ok, no problem. The thing I was worried about is that we rely on the order of the rules to find the right description and I don't think the generic dictionary guarantees the order of insertion is the order of enumeration but there's probably a low chance of it becoming an issue.

Reply:
@MikeFH actually you bring up a really good point that hadn't occurred to me. I don't think the mapping logic should rely the order in which keys are enumerated. Thinking about alternatives, I wonder if we could scrap the static mapping altogether and leverage the `System.Net.HttpStatusCode` enum instead. Something along the following lines:

```csharp
 var description = int.TryParse(statusCode, out int intValue)
      ? Enum.GetName(typeof(HttpStatusCode), intValue)
      : "Error";
```

Maybe some additional logic to split out the enum names into more natural text. For example, "BadRequest" becomes "Bad Request" etc. Thoughts?

Reply:
I looked a bit into it and I think it could be doable borrowing, for example, this method from Humanizer https://github.com/Humanizr/Humanizer/blob/main/src/Humanizer/StringHumanizeExtensions.cs#L26. But I noticed "429 Too many request" is missing from the HttpStatusCode enum. Maybe this could become a problem in the future if we want to support codes not in the enum.
In the end I wonder if we would not be better off hard coding all the codes from the enum plus others we need in the dictionary and simply get the description based on the key, removing the regex evaluation.